tools/RepoTasks/RepoTasks.Cmdlets.Tests/NewFormatPs1XmlCommandShould.cs

```diff
@@ -0,0 +1,247 @@
+﻿// ----------------------------------------------------------------------------------
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+namespace RepoTasks.Cmdlets.Tests
+{
+    using System;
+    using System.Linq;
+    using System.Management.Automation;
+    using System.Management.Automation.Runspaces;
+    using Xunit;
+    using System.IO;
+    using System.Collections.Generic;
+    using System.Xml.Linq;
+
+    public class NewFormatPs1XmlCommandShould
+    {
+        private const string CmdletName = "New-FormatPs1Xml";
+        private const string ExpectedAssemblyName = "RepoTasks.CmdletsForTest";
+
+        [Fact]
+        public void BuildPs1XmlFileAndSaveItOnDisk()
+        {
+            var initialSessionState = InitialSessionState.CreateDefault();
+
+            initialSessionState.Commands.Add(
```

Comment:
I am a little worried about polluting the environment and leaving commands arround.  We have an IDataStore interface here, with implementations for MockDataStore (for testing) and DiskDataStore (for runtime).  If you provide this at runtime (say through a static property, or the like) , then your tests could test this functionality without leaving files behind.  Let me know if this is too time-consuming to take up right now, as we could simply take this as test debt and schedule it later.

Reply:
@markcowl  
What if I just delete all the test artifacts, for example, in a final block?